e2e/testcases/reconciler_finalizer_test.go

```diff
@@ -0,0 +1,685 @@
+	)
+
+	rootRepo := nt.RootRepos[rootSyncNN.Name]
+	nsRepo := nt.NonRootRepos[repoSyncNN]
+	repoSyncPath := nomostest.StructuredNSPath(repoSyncNN.Namespace, repoSyncNN.Name)
+	deployment1NN := types.NamespacedName{Name: "helloworld-1", Namespace: repoSyncNN.Namespace}
+	deployment2NN := types.NamespacedName{Name: "helloworld-2", Namespace: repoSyncNN.Namespace}
+	namespace1NN := types.NamespacedName{Name: repoSyncNN.Namespace}
+
+	nt.T.Cleanup(func() {
+		cleanupMultiLevel(nt,
+			rootSyncNN,
+			repoSyncNN,
+			deployment1NN,
+			deployment2NN,
+			namespace1NN)
+	})
+
+	// Add deployment-helloworld-1 to RootSync
+	deployment1Path := nomostest.StructuredNSPath(deployment2NN.Namespace, "deployment-helloworld-1")
+	deployment1 := loadDeployment(nt, "../testdata/deployment-helloworld.yaml")
+	deployment1.SetName(deployment1NN.Name)
+	deployment1.SetNamespace(deployment2NN.Namespace)
+	rootRepo.Add(deployment1Path, deployment1)
+	rootRepo.CommitAndPush("Adding deployment helloworld-1 to RootSync")
+	nt.WaitForRepoSyncs()
+	nomostest.WaitForCurrentStatus(nt, kinds.Deployment(), deployment1.GetName(), deployment1.GetNamespace())
+
+	// Add deployment-helloworld-2 to RepoSync
+	deployment2Path := nomostest.StructuredNSPath(deployment2NN.Namespace, "deployment-helloworld-2")
+	deployment2 := loadDeployment(nt, "../testdata/deployment-helloworld.yaml")
+	deployment2.SetName(deployment2NN.Name)
+	deployment2.SetNamespace(deployment2NN.Namespace)
+	nsRepo.Add(deployment2Path, deployment2)
+	nsRepo.CommitAndPush("Adding deployment helloworld-2 to RepoSync")
+	nt.WaitForRepoSyncs()
+	nomostest.WaitForCurrentStatus(nt, kinds.Deployment(), deployment2.GetName(), deployment2.GetNamespace())
+
+	// Tail reconciler logs and print if there's an error.
+	// This is necessary because if the RootSync/RepoSync are deleted, the
+	// normal Cleanup won't be able to find the reconcilers.
+	// Start here to catch both the finalizer injection and deletion behavior.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go tailReconcilerLogs(ctx, nt, rootReconcilerObjectKey(rootSyncNN.Name))
+	go tailReconcilerLogs(ctx, nt, nsReconcilerObjectKey(repoSyncNN.Namespace, repoSyncNN.Name))
+
+	nt.T.Log("Enabling RootSync deletion propagation")
+	rootSync := fake.RootSyncObjectV1Beta1(rootSyncNN.Name)
+	err := nt.Get(rootSync.Name, rootSync.Namespace, rootSync)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	if setDeletionPropagationPolicy(rootSync, metadata.DeletionPropagationPolicyForeground) {
+		err = nt.Update(rootSync)
+		if err != nil {
+			nt.T.Fatal(err)
+		}
+	}
+	nomostest.WatchForObject(nt, kinds.RootSyncV1Beta1(), rootSync.GetName(), rootSync.GetNamespace(), []nomostest.Predicate{
+		nomostest.StatusEquals(nt, kstatus.CurrentStatus),
+		nomostest.HasFinalizer(metadata.ReconcilerFinalizer),
+	})
+
+	nt.T.Log("Disabling RepoSync deletion propagation")
+	repoSync := rootRepo.Get(repoSyncPath)
+	if removeDeletionPropagationPolicy(repoSync) {
```

Comment:
Shouldn't the RepoSync always have deletion propagation disabled at this point? I don't see anything before this point that should set it

Reply:
This is in prep for the follow up PR to enable it on all rsyncs for auto-cleanup, to simplify the test cleanup code.